src/main/java/net/fabricmc/loader/api/metadata/CustomValue.java

```diff
@@ -91,6 +112,20 @@
 		 * @return the value associated, or {@code null} if no such value is present
 		 */
 		CustomValue get(String key);
+
+		/**
+		 * Returns a sequential {@link Stream} with this iterable as its source.
+		 */
+		default Stream<Map.Entry<String, CustomValue>> stream() {
+			return StreamSupport.stream(this.spliterator(), false);
+		}
+
+		/**
+		 * Returns the set of keys in this custom value.
+		 */
+		default Set<String> keys() {
```

Comment:
This is good but I believe this should be abstract and hsndled in custom value impl. In the impl you could just Map.keySet()